### PR TITLE
unify plugin CLI commands

### LINKED ISSUE #512: Unify all flag in `plugin CLI` commands
## Checklist

- [ ] New Feature for the Service/Plugin (Vatz)
- [x] Enhancement (Vatz)
- [ ] others(etc. e.g, documentation, project policy management)

---

some of plugin cli command make users confuse to use 

such as 
```
 ./vatz plugin uninstall -p cpu_monitor
Error: unknown shorthand flag: 'p' in -p
Usage:
   plugin uninstall [flags]

Examples:
vatz plugin uninstall <pluginName>
```

```
 ./vatz plugin start                                                                                                   
2023-10-14T00:39:24-05:00 INF Start plugin   module=plugin
2023-10-14T00:39:24-05:00 ERR Get > QueryRowContext.Scan Error: sql: no rows in result set module=db
Error: sql: no rows in result set
Usage:
   plugin start [flags]

Examples:
vatz plugin start --plugin <pluginName> or vatz plugin start --plugin <pluginName> --args <arguments>

Flags:
  -a, --args string     Arguments
  -h, --help            help for start
  -l, --log string      Logfile
  -p, --plugin string   Installed plugin name

Global Flags:
      --config string   VATZ config file. (default "default.yaml")
      --debug           Enable debug mode on Log.
      --trace           Enable Trace mode on Log.

2023-10-14T00:39:24-05:00 ERR VATZ CLI command Error: sql: no rows in result set
 dongyookang DK 🍻   ~/ffplay/GolandProjects/xellos/vatz   ISSUE-511
```

Unify all plugin command's flags

### PR DESCRIPTION
**Pull request by xellos00**
### 1. Type of change

Please delete options that are not relevant.

- [ ] New feature
- [x] Enhancement
- [ ] Bug/fix (non-breaking change which fixes an issue)
- [ ] others (anything other than above)

---

**Related:** # (issue)
close #512 

**Summary**
As previously explained this, PR include to unify format of plugin cli command 

---

### CODE REVIEW

File: cmd/plugin.go [hunk 1]
```diff
@@ -209,26 +223,50 @@ var (
...
+			return err
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			pluginName := viper.GetString("disable_plugin")
+			pluginDir, err := config.GetConfig().Vatz.AbsoluteHomePath()
+			if err != nil {
+				return err
+			}
+
+			log.Debug().Str("module", "plugin").Msgf("enable installed plugin %s at %s", pluginName, pluginDir)
```
Comment (heejin-github) [inline cmd/plugin.go:266]: `disable installed plugin` seems to be correct log here.

Comment (kim201212) [reply cmd/plugin.go:266]: +1

File: cmd/plugin.go [hunk 2]
```diff
@@ -314,12 +352,25 @@ func createPluginCommand() *cobra.Command {
...
+		log.Error().Str("module", "plugin").Err(err)
+	}
+
 	cmd.AddCommand(statusCommand)
 	cmd.AddCommand(installCommand)
 	cmd.AddCommand(uninstallCommand)
 	cmd.AddCommand(startCommand)
 	cmd.AddCommand(stopCommand)
 	cmd.AddCommand(enableCommand)
+	cmd.AddCommand(disableCommand)
```
Comment (skonhwang) [inline cmd/plugin.go:373]: This is good to add commands more.  
I've refactored the command addition section using a loop. By storing all command objects in a slice and iterating over them, we can add each command to the root command with a single loop. This approach not only cleans up the code, making it more readable, but also simplifies any future modifications or additions to the command list.
This change might be a matter of personal preference, so it's ultimately up to your decision.

```
commands := []*cobra.Command{
    statusCommand,
    installCommand,
    uninstallCommand,
    startCommand,
    stopCommand,
    enableCommand,
    disableCommand,
    listCommand,
}

for _, cmd := range commands {
    rootCmd.AddCommand(cmd)
}
```

**Review by heejin-github — APPROVED** [review]
LGTM

**Comment by xellos00** [comment]
@here
github actions failed due to its authentication, I will fix it from issue. 
- 

here's code coverage test
```
 make coverage
?       github.com/dsrvlabs/vatz        [no test files]
?       github.com/dsrvlabs/vatz/manager/types  [no test files]
?       github.com/dsrvlabs/vatz/mocks  [no test files]
ok      github.com/dsrvlabs/vatz/cmd    2.662s  coverage: 19.1% of statements
ok      github.com/dsrvlabs/vatz/manager/api    0.459s  coverage: 0.0% of statements [no tests to run]
ok      github.com/dsrvlabs/vatz/manager/config 0.722s  coverage: 77.8% of statements
ok      github.com/dsrvlabs/vatz/manager/dispatcher     0.597s  coverage: 6.9% of statements
ok      github.com/dsrvlabs/vatz/manager/executor       0.858s  coverage: 65.5% of statements
ok      github.com/dsrvlabs/vatz/manager/healthcheck    1.408s  coverage: 44.4% of statements
ok      github.com/dsrvlabs/vatz/manager/plugin 7.093s  coverage: 51.5% of statements
ok      github.com/dsrvlabs/vatz/monitoring/prometheus  0.956s  coverage: 0.0% of statements
ok      github.com/dsrvlabs/vatz/rpc    2.103s  coverage: 65.0% of statements
ok      github.com/dsrvlabs/vatz/utils  1.596s  coverage: 8.5% of statements
 dongyookang DK 🐸   ~/ffplay/GolandProjects/xellos/vatz   ISSUE-512 ±
 
```

**Review by heejin-github — APPROVED** [review]
LGTM

**Review by skonhwang — APPROVED** [review]
LGTM

File: utils/utils_test.go [hunk 3]
```diff
@@ -32,3 +35,20 @@ func TestMakeUniqueValue(t *testing.T) {
...
+			fmt.Printf("Failed to kill process: %v\n", err)
+		}
+	}()
+
+	select {
+	case <-sigs:
+		// Test passed, signal received
+	case <-time.After(time.Second):
+		t.Error("Expected to receive signal, but did not")
+	}
```
Comment (rootwarp) [inline utils/utils_test.go:53]: What do you think adding assertion here to check having a signal.

File: cmd/plugin.go [hunk 4]
```diff
@@ -135,15 +133,31 @@ var (
...
+				if err != nil {
+					log.Error().Str("module", "plugin").Err(err)
+					return err
+				}
+				log.Info().Str("module", "plugin").Msgf("Plugin %s is successfully uninstalled from %s", args[0], pluginDir)
+			} else {
+				log.Error().Str("module", "plugin").Msgf("There's no plugin with the name %s from the installed plugin list. ", args[0])
+				log.Error().Str("module", "plugin").Msg("Please confirm plugin name again")
+			}
+
```
Comment (rootwarp) [inline cmd/plugin.go:160]: afaik, Golang prefers `if() return` statement rather than `if() else`.

File: cmd/plugin.go [hunk 5]
```diff
@@ -209,26 +223,50 @@ var (
...
 			pluginDir, err := config.GetConfig().Vatz.AbsoluteHomePath()
 			if err != nil {
 				return err
 			}
 
-			log.Debug().Str("module", "plugin").Msgf("enable installed plugin %s at %s", args[0], pluginDir)
+			log.Debug().Str("module", "plugin").Msgf("enable installed plugin %s at %s", pluginName, pluginDir)
+
+			mgr := plugin.NewManager(pluginDir)
+			err = mgr.Update(pluginName, true)
```
Comment (rootwarp) [inline cmd/plugin.go:242]: The name `Update` seems quite ambiguous, what do you think `SetEnable` for the new one?(but I'm not sure doing this in this PR is a good choice.)

File: utils/utils_test.go (hunk 3 above)
Comment (xellos00) [reply utils/utils_test.go]: @xellos00 
Added

File: cmd/plugin.go (hunk 4 above)
Comment (xellos00) [reply cmd/plugin.go]: @rootwarp 
fixed

File: cmd/plugin.go (hunk 5 above)
Comment (xellos00) [reply cmd/plugin.go]: @rootwarp 
fixed!!

**Review by rootwarp — APPROVED** [review]
LGTM

**Pull request merged**